File: app/src/main/java/com/example/waveplayer/media_controller/Song.java
package com.example.waveplayer.media_controller;

import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

import java.io.Serializable;

@Entity(tableName = "songs")
public class Song implements Comparable<Song>, Serializable {

    @PrimaryKey
    public final Long id;

    @ColumnInfo(name = "title")
    public final String title;

    @Ignore
    private boolean selected;

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public Song(Long id, String title){
        this.id = id;
        this.title = title;
    }

    @Override
    public int compareTo(Song o) {
        return title.compareTo(o.title);
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return obj instanceof Song && id.equals(((Song) obj).id);
    }

    @Override
    public int hashCode() {
        return System.identityHashCode(this);
    }

}

File: app/src/main/java/com/example/pinky_player/random_playlist/Song.kt
package com.example.pinky_player.random_playlist

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import java.io.Serializable

@Entity(tableName = "songs")
class Song(@field:PrimaryKey val id: Long, @field:ColumnInfo(name = "title") val title: String) : Comparable<Song>, Serializable {
    @Ignore
    private var selected = false
    fun isSelected(): Boolean {
        return selected
    }

    fun setSelected(selected: Boolean) {
        this.selected = selected
    }

    override fun compareTo(other: Song): Int {
        return title.compareTo(other.title)
    }

    override fun equals(other: Any?): Boolean {
        return other is Song && id == (other).id
    }

    override fun hashCode(): Int {
        return System.identityHashCode(this)
    }
}